process/smartContract/process.go

```diff
@@ -58,37 +60,41 @@ func NewSmartContractProcessor(
 	coordinator sharding.Coordinator,
 	scrForwarder process.IntermediateTransactionHandler,
 	txFeeHandler process.TransactionFeeHandler,
+	economicsFee process.FeeHandler,
 ) (*scProcessor, error) {
-	if vmContainer == nil || vmContainer.IsInterfaceNil() {
+	if check.IfNil(vmContainer) {
```

Comment:
you might consider extracting all these checks to a function called `checkArgumentsForNil`

Reply:
done.